### ISSUE TITLE
OpenBSD (-current) Test fails

### ISSUE THREAD

**Issue by gonzalo-** [syncthing/syncthing#5077]
Hello,

I am having some problems with the "tests" on syncthing on >=0.14.47

Log attached.

Thanks!

### Version Information

Syncthing Version: >=0.14.47
OS Version: OpenBSD-current (probably also in 6.3)
[syncthing.txt](https://github.com/syncthing/syncthing/files/2226986/syncthing.txt)

**Comment by gonzalo-** [comment]
Hello, any news about this?

thanks

**Comment by vext01** [comment]
I can confirm that this is still a problem for version 0.14.50 running OpenBSD-6.4-beta.

Test log attached.

[test.log](https://github.com/syncthing/syncthing/files/2441375/test.log)

**Comment by AudriusButkevicius** [comment]
@imsodin how would one debug this? I guess logging won't log much about this?

**Comment by imsodin** [comment]
Running with `fs` and `watchaggregator` (`STTRACE=fs,watchaggregator`) set may provide some hints. And if it is still flaky (?) also provide those logs from a successful test run.

**Comment by vext01** [comment]
Here's a log with that environment set.

[test.log.gz](https://github.com/syncthing/syncthing/files/2445636/test.log.gz)

Let me know if there's anything else I can provide.

**Comment by AudriusButkevicius** [comment]
Uff, the log is convoluted, it's not clear which lines belong to which test :(

```
2018/10/04 09:57:24.465273 folder_sendrecv.go:1749: INFO: Puller (folder "default" (default), file "symlink/testdir"): traverses q: traverses symlink: symlink
2018/10/04 09:57:24.465296 folder_sendrecv.go:1749: INFO: Puller (folder "default" (default), file "symlink/testsyml"): traverses q: traverses symlink: symlink
2018/10/04 09:57:24.465321 folder_sendrecv.go:1749: INFO: Puller (folder "default" (default), file "symlink/testfile"): traverses q: traverses symlink: symlink
``` 

But I don't think this could be it?

**Comment by imsodin** [comment]
There is not just `TestWatchIgnore` failing (which I believe did not log anything, my proposition was BS, sorry), but also `TestInternalScan`, `TestIssue4903` and `TestParentDeletion`. All of the latter are more important than the watch error. So I'd suggest navigation to `./lib/model` and running `STTRACE=model go test -race -v -run TestInternalScan` (same for other to tests) to see what's going on there. And maybe move this over to the forum (?).

**Comment by vext01** [comment]
OK, I have all the test outputs via the method outlined by @imsodin above.

If it's OK, I'd rather post the issues one per issue on github.

**Comment by vext01** [comment]
I've raised the first of the failures in #5243.

**Comment by vext01** [comment]
Sorry for the duplication (my bad).

Here's the info about this failure:
```
fremen% STTRACE=fs go test -v -run TestWatchIgnore
=== RUN   TestWatchIgnore
2018/10/05 14:33:26.878085 logfs.go:73: DEBUG: basicfs_watch_test.go:363 basic /home/porter/go/src/github.com/syncthing/syncthing/lib/fs/testdata MkdirAll ignore 493 <nil>
2018/10/05 14:33:26.878817 logfs.go:133: DEBUG: basicfs_watch_test.go:371 basic /home/porter/go/src/github.com/syncthing/syncthing/lib/fs/testdata Watch ignore {ignore/ignored  true} false <nil>
2018/10/05 14:33:26.878887 logfs.go:73: DEBUG: basicfs_watch_test.go:339 basic /home/porter/go/src/github.com/syncthing/syncthing/lib/fs/testdata MkdirAll ignore 493 <nil>
2018/10/05 14:33:26.879139 logfs.go:43: DEBUG: basicfs_watch_test.go:342 basic /home/porter/go/src/github.com/syncthing/syncthing/lib/fs/testdata Create ignore/file {0xc00000c068 ignore/file} <nil>
2018/10/05 14:33:26.879189 logfs.go:73: DEBUG: basicfs_watch_test.go:339 basic /home/porter/go/src/github.com/syncthing/syncthing/lib/fs/testdata MkdirAll ignore 493 <nil>
2018/10/05 14:33:26.879387 logfs.go:43: DEBUG: basicfs_watch_test.go:342 basic /home/porter/go/src/github.com/syncthing/syncthing/lib/fs/testdata Create ignore/ignored {0xc00000c070 ignore/ignored} <nil>
2018/10/05 14:34:26.886073 logfs.go:103: DEBUG: basicfs_watch_test.go:386 basic /home/porter/go/src/github.com/syncthing/syncthing/lib/fs/testdata RemoveAll ignore remove /home/porter/go/src/github.com/syncthing/syncthing/lib/fs/testdata/ignore: directory not empty                                                    
--- FAIL: TestWatchIgnore (60.01s)
    basicfs_watch_test.go:382: Timed out before receiving all expected events
FAIL
exit status 1
FAIL    github.com/syncthing/syncthing/lib/fs   60.021s
```

This was using the git version:`f528923d1ef224a07d58fea21b0dca14ca18ba78`

**Comment by AudriusButkevicius** [comment]
This still fails on the VM I was given, but I assume because of inotify simply not working on openbsd.

**Comment by AudriusButkevicius** [comment]
Ref https://github.com/rjeczalik/notify/issues/172

**Comment by AudriusButkevicius** [comment]
Can we just disable the inotify tests on OpenBSD and get on with our lives?

**Comment by calmh** [comment]
Sounds like the watcher works (going by the talk in the other issue), just that recursive watches take some time to take effect. This shouldn't really affect Syncthing other than by the test failure, as we'll do a recursive scan of the path we get a notification for and see any additions below it as well. And they get watched too, just with a short delay?

**Comment by vext01** [comment]
Looking at Audrius's experimentation on the notify issue, it looks like the delay incurred is a matter of seconds. Could the test be fixed by simply waiting longer on OpenBSD?

> Generally disable watcher on openbsd i assume?

What are the consequences of disabling the watcher?

**Comment by AudriusButkevicius** [comment]
> Could the test be fixed by simply waiting longer on OpenBSD?

No, because this waiting needs to happen in ths inotify library and not our tests.

Also, I always have a sudden urge to throw up while adding sleeps to fix the tests.

I think we disable the tests, declare inotify support on openbsd "best effort but might be broken" and move on.

**Comment by vext01** [comment]
Does syncthing operate fine without inotify?

Cheers

**Comment by AudriusButkevicius** [comment]
It should. There are periodic rescans to fill the gap, and also us knowing that some directory was created would force syncthing to traverse the whole tree down anyway.

**Comment by vext01** [comment]
In that case, I think your proposal sounds good. But perhaps add a comment
saying why we skip the test?

Once again, thanks for looking into this. I'm most grateful.

**Comment by vext01** [comment]
> perhaps add a comment saying why we skip the test?

One more thing. Perhaps in that comment link to the inotify issue.
